.github/workflows/run_pytest.yml

```
@@ -0,0 +1,27 @@
+name: Run pytest
+
+on:
+  pull_request:
+    branches:
+      - main
```

```suggestion
      - main
      - dev
```


Lets add this now, even though we haven't branched it yet (waiting until necessary)